src/openforms/products/admin/product.py

```diff
@@ -1,8 +1,24 @@
+from django import forms
 from django.contrib import admin
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
 
 from ..models import Product
 
 
+class ProductAdminForm(forms.ModelForm):
+    class Meta:
+        model = Product
+        fields = "__all__"
+
+    def clean_price(self):
+        value = self.cleaned_data["price"]
+        if value <= 0:
+            raise ValidationError(_("The value must be greater than 0."))
+
+        return value
```

Comment:
I think it'd be nicer to create a validator based on `MinValueValidator` from `django.core.validators` and apply that on the model field through `validators=[NonZeroValidator()]`, what do you think?